### ISSUE TITLE
`undefined/undefined` is wrong...

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#16643]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

the details for a not found resource reflected `undefined/undefined:`
```
Resource not found in cluster: undefined/undefined:cert-manager-startupapicheck
```
**To Reproduce**

1. let argocd manage a cert-manager install
2. let argocd upgrade cert-manager install
3. while argocd is triggering the cert-manager upgrade, a service account for `cert-manager-startupapicheck` will appear and can be clicked (?)
4. click it
5. depending on (hand waving) the `cert-manager-startupapicheck` will probably already be gone? so a confusing error will be rendered by argocd

note that for a while longer i could ask argocd to filter for service accounts and see it, but i couldn't click it to actually show the item... and it eventually stopped appearing...

**Expected behavior**

information about expect objects should be stored in the request pointer so that they can be filled in in cases when they no longer exist instead of asking the ui to imagine them after the fact

**Screenshots**

![image](https://github.com/argoproj/argo-cd/assets/2119212/7a447ecd-11de-40f6-ae4a-385d19c55128)

**Version**

```shell
{
    "Version": "v2.8.3+77556d9",
    "BuildDate": "2023-09-07T16:05:43Z",
    "GitCommit": "77556d9e64304c27c718bb0794676713628e435e",
    "GitTreeState": "clean",
    "GoVersion": "go1.20.6",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v5.1.0 2023-06-19T16:58:18Z",
    "HelmVersion": "v3.12.1+gf32a527",
    "KubectlVersion": "v0.24.2",
    "JsonnetVersion": "v0.20.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by faust64** [comment]
Not sure it's related ... but as of upgrading ArgoCD (from 2.7.14 to 2.8.4), I'm seeing similar messages.
For just any object, eg:

> Resource not found in cluster: v1/ConfigMap:cronjob-tektonruns-garbage-collect-7h57hctc55
> Please update your resource specification to use the latest Kubernetes API resources supported by the target cluster. The recommended syntax is v1/ConfigMap:cronjob-tektonruns-garbage-collect-7h57hctc55

Object is present in cluster though.
Application shows healthy, it's not OutOfSync.
I would just get that message when looking at individual resources "Live Manifest" in ArgoCD console.

This affects all objects in my applications: Services, ConfigMaps, NetworkPolicies ...
Although I don't have that "undefined/undefined" thing ...

Still testing 2.8 / first installation I'm upgrading. I can confirm this is a not an issue, with exact same applications, running 2.7.14
Using that OpenShift GitOps operator lifecycling my ArgoCD -- ArgoCD object did not change in a while, as we did proceed with sso/dex reconfiguration when upgrading from 2.6 to 2.7.

**Comment by jsoref** [comment]
Sorry, that's a different issue. If I were you, I'd join slack to debug and then file a ticket (feel free to reference it here so people can find it if they stumble on your note here instead of the new ticket).

**Comment by faust64** [comment]
I am.
Still I'm wondering: does the live manifest view works at all?
You may be having two problems here.

**Comment by jsoref** [comment]
Yes, live manifest works fine for everything else.

**Comment by ku524** [comment]
I am also experiencing the same issue. My version of ArgoCD is 2.9.1. In my case, it was a pod, and I couldn't delete this pod through any method supported by ArgoCD. Note that the actual pod had already been deleted. However, removing the owner allowed me to delete this pod.
<img width="1212" alt="스크린샷 2024-01-09 오후 1 48 57" src="https://github.com/argoproj/argo-cd/assets/29232807/82232029-fa60-45b9-985b-730a0e1b1bac">

**Comment by pargupta12** [comment]
@jsoref: is your issue resolved? I am also facing same issue and getting error message something like 
"Resource not found in cluster: undefined/undefined:ly-job-a67f890-postsync-1707682950" 

if you have some workaround. Please suggest.Thanks

**Comment by jsoref** [comment]
No, it isn't resolved. I just deal with it.

**Comment by bofaalsarah** [comment]
We just upgraded our ArgoCD from 2.7.6 to 2.10.1 and we are facing a similar issue

`Resource not found in cluster: v1/ConfigMap:bitbucket-runner-configs-59t96976td
Please update your resource specification to use the latest Kubernetes API resources supported by the target cluster. The recommended syntax is v1/ConfigMap:bitbucket-runner-configs-59t96976td`

**Comment by jsoref** [comment]
Sorry, that's a different issue. If I were you, I'd join slack to debug and then file a ticket (feel free to reference it here so people can find it if they stumble on your note here instead of the new ticket).

**Comment by bofaalsarah** [comment]
I think it's still related because when I look into the live manifest tab of the pod, I am getting 

`Resource not found in cluster: undefined/undefined:bitbucket-runner-7c87665b77-m886p
`

**Comment by jsoref** [comment]
Oh, sorry, I didn't see that in what you pasted earlier. Yes, that's this problem :).

**Comment by jsoref** [comment]
Fwiw, this is more about error reporting than fixing the underlying problem causing any given error.

**Comment by jsoref** [comment]
Fwiw, this appears to have been introduced by @ashutosh16 in #13031 -- it's odd, because there are screenshots of it working there, so, presumably the feature works _some_ of the time.

The relevant code is:
https://github.com/argoproj/argo-cd/blob/891d08930470675c8851909254b0b5bd45eb79f0/ui/src/app/applications/components/application-node-info/application-node-info.tsx#L240-L241

**Comment by LS80** [comment]
Are you missing the [cluster rbac](https://github.com/argoproj/argo-cd/tree/master/manifests/cluster-rbac/server) for the `server` component?

**Comment by cmarasescu** [comment]
I had the same issues myself after install ArgoCD v2.10.1.
I've followed the official install instructions (https://argo-cd.readthedocs.io/en/stable/getting_started/, with the only difference that I've use a custom namespace for installation (argocd-test)
```
kubectl create ns argocd-test
kubectl apply -n argocd-test -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/core-install.yaml
```

1. Issue 1: live manifest was showing errors for all resources, even tough the resources were deployed successfuly. In example, the errors for a configMap:

```
Resource not found in cluster: v1/ConfigMap:public-ingress-controller-test 
Please update your resource specification to use the latest Kubernetes API resources supported by the target cluster. 
The recommended syntax is v1/ConfigMap:public-ingress-controller-test
```
2. Issue 2: I was not able to delete resources from ArgoCD GUI - in example, while trying to delete a confgiMap, I got the following error:
```
Unable to delete resource: error deleting resource: configmaps "public-ingress-controller-test" is forbidden: 
User "system:serviceaccount:argocd-test:argocd-server" cannot delete resource "configmaps" in API group "" in the namespace "lu-ingress-controller"
```

Resolution:
- It seems that even tough I've installed ArgoCD in a custom namespace, the installation file used above - [https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/core-install.yaml](https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/core-install.yaml) - had hardcoded the `argocd` namespace for the `argocd-server` ClusterRoleBinding 
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/component: server
    app.kubernetes.io/name: argocd-server
    app.kubernetes.io/part-of: argocd
  name: argocd-server
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: argocd-server
subjects:
- kind: ServiceAccount
  name: argocd-server
  namespace: argocd
```

The solution in my case  was to adapt the `argocd-server` ClusterRoleBinding to also include the `argocd-server` serviceAccount from my custom namespace `argocd-test`. Basically, under subjects I've added the following 2 lines:
```
subjects:
- kind: ServiceAccount
  name: argocd-server
  namespace: argocd-test
```

**Comment by ppodevlabs** [comment]
i'm facing the same issue. I've deployed argocd 2.10.2 in cluster setup with the applications in any namespace, i've made the proper configuration and deployed https://github.com/argoproj/argo-cd/tree/master/manifests/cluster-rbac too.

But my application keeps syncing
<img width="1614" alt="Screenshot 2024-03-15 at 07 08 10" src="https://github.com/argoproj/argo-cd/assets/7081272/02023a4c-7e5f-4c4f-9883-229ba9b2c642">


Actually the deployment/ingres/networkpolicy are deployed and healthy but argo does not seems to detect them.

**Comment by cooervo** [comment]
I was having this issue in our helm values file we had some custom config for `server.clusterRoleRules.rules` and had `server.clusterRoleRules.enabled: true` that was overriding default config notice: 

```
{{- if .Values.createClusterRoles }}
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: {{ include "argo-cd.server.fullname" . }}
  labels:
    {{- include "argo-cd.labels" (dict "context" . "component" .Values.server.name "name" .Values.server.name) | nindent 4 }}
rules:
  {{- if .Values.server.clusterRoleRules.enabled }}
    {{- toYaml .Values.server.clusterRoleRules.rules | nindent 2 }}
  {{- else }}
  - apiGroups:
      - '*'
```

had to remove the custom rules and `server.clusterRoleRules.enabled: false`  same as in default argocd values file. and now I can see all resources just fine

**Comment by andrii-korotkov-verkada** [comment]
ArgoCD versions 2.10 and below have reached EOL. Can you upgrade and let us know if the issue is still present, please?

**Comment by andrii-korotkov-verkada** [comment]
There were ~1600 open bugs, so at least some managing is needed. I know my comment is less than ideal, but I don't see a better way to prioritize at this point. Also, there's no easy way to tell if the issue got fixed. For example, perf improvements in 2.13 for getting application tree can fix a whole class of refresh-related issues, but there's no 100% way to know.

I'll try to improve the situation with issues that don't get as much attention, but it'd take time and require multiple steps. For now sure, I won't ping your issues asking to reproduce.

**Comment by jsoref** [comment]
Because we happened to trip on it today, here's a screenshot from v2.12.4+27d1e64:

<img width="397" alt="image" src="https://github.com/user-attachments/assets/c4f1ecda-5abc-4ec5-9757-a229eec185b6">

**Comment by ap0phi5** [comment]
Same here. We are also seeing this exact issue.
v2.12.3+6b9cd82

**Comment by andrii-korotkov-verkada** [comment]
I think `undefined/undefined` can be fixed by using `props.node` fields, e.g. this one https://github.com/argoproj/argo-cd/blob/6d8d32f170c9db33cd34029cc554afeb335a5faf/ui/src/app/applications/components/application-node-info/application-node-info.tsx#L101. I don't know if there are any issues with that.

I don't yet understand why these nodes are shown at all and whether it's acceptable to hide them, e.g. if `props.controlled` is not there.

**Comment by UriZafrir** [comment]
same error here..

![image](https://github.com/user-attachments/assets/85fda1d3-e741-4fb3-ac8c-8478f724cbfc)

**Comment by jsoref** [comment]
@UriZafrir: that's a different bug... this is limited for the case of `undefined`/`undefined`.

**Comment by adrianmiron** [comment]
I confirm the issue is still there, using `v2.13.1+af54ef8`

**Comment by JohnnyJayJay** [comment]
I had an issue that _looked_ very similar (could not find resource in cluster: undefined/undefined:... etc.) but was fundamentally different. In my case the resources were created correctly by the Argo CD application-controller because its `ServiceAccount` had the required permissions, but the argocd-server `ServiceAccount` in charge of serving the UI had no permissions in the namespace the resources were created in. This can happen if you install Argo CD in a way that doesn't automatically give it admin permissions in all of the (managed) namespaces, which was the case for me (installation via the OpenShift operator in a restricted environment).

@LS80 already pointed in this direction in an earlier [comment](https://github.com/argoproj/argo-cd/issues/16643#issuecomment-1980435372), just wanted to make it clear for other people who come across this. If it's simply a permissions issue, it can be fixed by creating an appropriate `RoleBinding` or `ClusterRoleBinding` for the argocd-server ServiceAccount in the namespace(s) where resources are falsely reported as missing.

Again, this is not the same issue as originally reported here, but I think it still might be useful to point it out here.

**Comment by adrianmiron** [comment]
@JohnnyJayJay Yeah, i should have mentioned, my issue is the one with resources that no longer exist on the cluster ( pods in my case ). I just created some nginx  pods with `kubectl run` with the right name in the right namespace, argo managed to complete it's delete command ( labels and other selectors were no longer relevant ) and the issue went away :)

**Comment by Gamote** [comment]
I’d like to share an alternate setup perspective that might help anyone facing this issue.

### Context

We ran into the exact same RBAC error, first on the Bitnami Argo CD chart and then on the official chart:

1. **Bitnami chart**  
   - Enabled cluster-admin access by default (`server.clusterAdminAccess: true`).  
   - Still saw the permission errors.

2. **Official chart**  
   - Manually added a `ClusterRole` for `argocd-server` (as described by @JohnnyJayJay).
   - Errors persisted.

It turned out that the issue was that our setup has one dedicated **infrastructure cluster** (running Argo CD) and several **product clusters** (all orchestrated by ArgoCD through [GKE Fleet](https://cloud.google.com/kubernetes-engine/docs/fleets-overview) + [Connect Gateway](https://cloud.google.com/kubernetes-engine/enterprise/multicluster-management/gateway/using)). A single Kubernetes `ClusterRole` on the Argo CD cluster only fixes permissions for apps in that same cluster—not across separate clusters. Which now, makes sense. :)

### Solution

> **Note:** This focuses on GCP, but the same pattern should work on other clouds that support workload identity–style federation. We also implemented this solution fully in Terraform, but in order to keep it simple I used the raw resources.

> **Prerequisite:** Your GKE cluster must have [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) enabled.

1. **Create a Google Service Account (GSA)**
   *Give it **Reader** access (`roles/container.clusterViewer` is enough) on every product cluster.*

2. **Create a Kubernetes Service Account (KSA) for `argocd-server`** in the infra cluster.

   ```yaml
   apiVersion: v1
   kind: ServiceAccount
   metadata:
     name: argocd-server
     annotations:
       iam.gke.io/gcp-service-account: <GSA_EMAIL>
   ```

3. **Bind the KSA to the GSA** (Workload Identity):

   ```bash
   gcloud iam service-accounts add-iam-policy-binding <GSA_EMAIL> \
     --member "serviceAccount:<PROJECT_ID>.svc.id.goog[<NAMESPACE>/argocd-server]" \
     --role roles/iam.workloadIdentityUser
   ```

4. **Redeploy Argo CD** so the `argocd-server` Deployment uses the new KSA.

After this change, Argo CD had the correct read-only permissions across all clusters, and the RBAC errors disappeared! ✨

**Comment by jsoref** [comment]
Fwiw, 
- https://github.com/argoproj/argo-cd/pull/25066 addresses the `recommended syntax` item raised here.

**Comment by clementpsp** [comment]
Apologies for bumping this issue, has the fix [#25068](https://github.com/argoproj/argo-cd/pull/25068) been released to an ArgoCD version?

I don't see the fix code [in the v3.2.5 tag](https://github.com/argoproj/argo-cd/blob/v3.2.5/ui/src/app/applications/components/application-node-info/application-node-info.tsx), nor a mention in any changelog entry since that PR was merged in October.

**Comment by jsoref** [comment]
https://github.com/argoproj/argo-cd/commit/0d0cec6b66c906b1dda6784637e1dfb2ada328f1 shows it's in [v3.0.0-rc1](https://github.com/argoproj/argo-cd/releases/tag/v3.3.0-rc1) and newer. I'd expect it to be in v3.0.0 when that releases (I don't know the timeline).
